sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Mode.scala

```diff
@@ -74,16 +85,25 @@ case class Mode(
     if (buffer.isEmpty) {
       return null
     }
-
+    val collationAwareBuffer = child.dataType match {
```

Comment:
here would be a good place for a larger comment explaining the logic we used for collation awareness in `Mode`

Reply:
@uros-db I have added the requested comment. let me know if there is any way to improve what I added.